src/bgen/Generator.cs

```diff
@@ -585,19 +585,19 @@ public TrampolineInfo MakeTrampoline (Type t)
 
 		if (mi.ReturnType.IsArray && TypeManager.IsWrappedType (mi.ReturnType.GetElementType ())) {
 			returntype = NativeHandleType;
-			returnformat = "return NSArray.FromNSObjects({0}).GetHandle ();";
+			returnformat = "return Runtime.RetainAndAutoreleaseNSObject (NSArray.FromNSObjects({0}));";
```

Comment:
We'll have to make `Runtime.RetainAndAutoreleaseNSObject` public (it's currently internal), since it can now be called from generated user code.

Also add EditorBrowsable=Never on it.